From e31c41c02a173e10ad1ef6ca1d475366f818d3cb Mon Sep 17 00:00:00 2001
From: Nikita Travkin <nikita@trvn.ru>
Date: Wed, 5 May 2021 17:48:33 +0500
Subject: [PATCH 12/24] FIXME: arm64: dts: qcom: apq8016-samsung-gt510wifi: Add
 fuelgauge

gt5 devices use additional MAX77849 PMIC to handle battery, charging
and usb switching. Fuelgauge from this PMIC has a separate i2c line.

Signed-off-by: Nikita Travkin <nikita@trvn.ru>
---
 .../dts/qcom/msm8916-samsung-gt5-common.dtsi  | 44 +++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
index b56974e361..05dee27c3d 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
@@ -64,6 +64,34 @@ hall-sensor {
 		};
 	};
 
+	i2c-fuelgauge {
+		compatible = "i2c-gpio";
+		sda-gpios = <&msmgpio 14 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&msmgpio 15 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&fuelgauge_i2c_default>;
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		fuelgauge@36 {
+			// FIXME: maxim,max77849-battery
+			compatible = "maxim,max17047";
+			reg = <0x36>;
+
+			maxim,rsns-microohm = <10000>;
+			maxim,over-heat-temp = <600>;
+			maxim,over-volt = <4400>;
+
+			interrupt-parent = <&msmgpio>;
+			interrupts = <121 IRQ_TYPE_EDGE_FALLING>;
+
+			pinctrl-names = "default";
+			pinctrl-0 = <&fg_int_default>;
+		};
+	};
+
 	// FIXME: Use extcon device provided by MUIC driver when available
 	usb_vbus: usb-vbus {
 		compatible = "linux,extcon-usb-gpio";
@@ -283,6 +311,22 @@ l18 {
 };
 
 &msmgpio {
+	fg_int_default: fg-int-default {
+		pins = "gpio121";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	fuelgauge_i2c_default: fuelgauge-i2c-default {
+		pins = "gpio14", "gpio15";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
 	gpio_keys_default: gpio-keys-default {
 		pins = "gpio107", "gpio109";
 		function = "gpio";
-- 
2.25.1

